### ISSUE TITLE
force promote b0963378 in order to end a week's worth of dry spells?

### ISSUE THREAD

**Issue by hydrogenpi** [gcp/leela-zero#1091]
whats going on folks, almost a week no net promotions, we get a 54.74% (b0963378, same network size) and it dosen't round up and push through? What ever happend to talks of lowering the gating threshold as to not waste community time ? Sure a 51% might not be worth it and you'd have to run 10000  games to be sure it isn't a false postitive, but at 54.74% come on!!! I feel like I been on nofap for a year... is the quality over quantit\y doctrine being implemented? How about run it to 600 games and if it stays at 54% then promote it regardless? At this level 55% gating is too high and counterdeminishing, up the MCTS, boost quality, lower gating, or at least round up and give very close ones a pass esp. when there has been no rain forever. If we had a stronk 20 net or 192 filter contender up had already ready to take over then this won't be so much of an issue but it appears that isn't the case either. So we need some quantitative easing here. 54 is the new 55. we can't keep lowering the interest rate forever but at least this buys some time.

**Comment by tterava** [comment]
Dry spells are bound to happen in projects like this with significant amount of random involved. The 55% winrate over 400 games is chosen for two reasons: 1. It's what AGZ used and 2. It gives 95% confidence that the new network actually is better.

Lowering the criteria wouldn't mean we get stronger networks more quickly. Most likely we would get more false positives and we'd think we're making progress when we're not. You can always download the network manually and use it without it being an official promotion as well.

GCP has tested quite a few new boostraps lately, so he's probably planning to up the network size. I'm sure he has a plan going forward so we'll probably get a new architecture somewhere in the near future.

**Comment by hydrogenpi** [comment]
I checked historical patterns, since Nov of 2017, if I not mistaken, this six day dry spell is the single longest one yet...

**Comment by bldght** [comment]
I want to know what the abandoned  network actually learned? Is it really bad?I think there is no really "bad" network.  Since there are ways to improve the network now, why not try the AlphaZero method?

**Comment by Friday9i** [comment]
It is now at 54.95%... Just one remaining win and it is a PASS, fingers crossed!
Alternatively, lowering the threshold (52 or 53%?) or taking a new approach with a pool of good networks (see my previous post for details) would solve this painful issue.

**Comment by MaxMaki** [comment]
It's extremely difficult to know what exactly a 4% winrate improvement looks like in "things learnt", but nothing stops you from downloading the network and trying it, or just looking at the heatmaps.

Whether or not the network should be manually promoted is up to the project management, i personally seriously doubt opinions there will be changed by "it's my opinion that..." type posts.

@hydrogenpi There have been several 200-250k stalls. If i remember correctly there was a 5x64 network that actually got an improvement when over 250k games had been played on that.  It's not a question about time, it's a question about how many games have been generated. Every single time an old network has neared its limit it has been like this, it's just something you need to accept. The last parts of squeezing out strength will take a long time, and then hopefully a bootstrap will work well.
GCP has already stated many times his intentions on when and how he'll lower learning rates, and those have always been based on how many games in the training window are from one window. I'm guessing from earlier posts that we still have 1 learning rate reduction to go before we start switching to a larger network (just my guess), and since the current net is nearing 200k games, hopefully we'll see it soon. 
Or maybe a new release with the fpu reduction being removed at the root node will be published first?

**Comment by l1t1** [comment]
suggest to dispatch more matchs to it to prove itself

**Comment by ThePokerDude** [comment]
Sounds like good idea.
Maybe it's possible to implement some automatism which schedules additional match games when a certain win percentage is achieved (53% or more or something like this)

**Comment by pheasant75** [comment]
With more games played, you will be more confident about the winning rate.  Therefore, the threshold of 55% can be lowered.  Has anyone done the math to see what's the threshold when there are 420 games?

**Comment by StanTraykov** [comment]
Is that satire? A plan to Make Leela Zero Great Again?

**Comment by Cabu** [comment]
b0963378 is a bigger network than the current one. I would like to let some more time in hope of a last better one before going bigger :)

**Comment by hydrogenpi** [comment]
@Friday9i
54.95% is for all intents and purposes 55%. The problem with not manually promoting it when its already been a week without new nets is that there is the possibility that by missing this chance, there will be another week of waiting for a 55% to come around. Whereas had this been promoted it would indeed be a stronger net (seriously what is the different of 54.95% and 55.01% at >400 games?!) and we all know that with stronger net means stronger self play games means stronger training means stronger new net... and it gets LZ out of the current viscial rut cycle of a loop that it is in that it seems it can't get out of... I thought with all the talk about lowering gating, or even doing away with gating, that when you go a week without a new net (all time record in length btw) and a 54.95% comes along and it doesn't get manually promoted, thats a bit curious.

**Comment by hydrogenpi** [comment]
@MaxMaki, its not a 4% its a 4.95%... for all intents and purposes it is a 55% we are talking about not a 54%.

**Comment by ThorAvaTahr** [comment]
This discussion is interesting from the point of view of how the community feels the need for progress in order to remain engaged with the project.

From a purely technical point of view I would take the side of not changing the rules of the experiment half-way. However we are continuously doing that anyway (from all the changes pulled only a few were actual bugs, while a lot were actually changing the experiment. Also to please my curiosity I would like to do anything to get the strongest possible network. Anyway we depend on keeping the community engaged. In principle I would be willing to accept some small deviations from our experiment for that, however, I fear that if only a week's stall is already testing the patience of our users, we will be in for a lot of trouble in the near future. Because the next network may take a couple of weeks, and game generation will only be slower with larger networks.

On the other hand, I am a bit worried that we have trouble bootstrapping the larger networks and after 6 million games we are stuck at a level quite far from the top of the CGOS list. I had expected we would either be at a higher level by now, or have a steeper progress

**Comment by hydrogenpi** [comment]
@Cabu you are incorrect. b0963378 is the exact same 10 block and with the exact same 128 filters and its download size is the same as the rest of them. It is identicially sized network as all current networks.

**Comment by hydrogenpi** [comment]
@ThorAvaTahr , I apologize if I suggested that I speak for the community because obviously I do not. I'm sure there are those in the community that wish to be even more pure and frown upon even the net2net that was done.

**Comment by ThorAvaTahr** [comment]
@hydrogenpi My apologies if I suggested that you speak for the community. I would rather say you speak from the community (same as me) and perhaps voice what a part of that community thinks as well .

From my point of view the competition element is not so relevant. The experiment counts, as does the result: open knowledge of learning go and a strong network for public use.

**Comment by hydrogenpi** [comment]
@ThorAvaTahr  so I guess this is something only the dev would have more visibility into but its relevancy depends in part on the make up of the users who's clients contribute cycles to the project. Are most of them of the academic type that want to see a recreation even if Google were to release all code and weights to AGZ tomorrow? Or even if a sizeable portion of the community is contributing because they want a better Go AI bot, (since Deepmind withheld the weights) and not because so much they are interested in the AI or academic angle, then that could be a problem for a project that relies on distrubtion contribution of computing.

**Comment by MaxMaki** [comment]
@hydrogenpi There were plenty of people producing games when we were stuck in the beginning due to bugs and were not even able to reach 30 kyu, and there will be plenty of people producing games after you've grown frustrated with the project not conforming to your opinions and ideas of what the project should be. I think GCP has stated to goals of this project many times, and it has never been to create the strongest possible bot, nor has it been there to get there quicker than anyone else. If those are the things important to you, i think you'll be dissapointed.

I think you are being rather forcing in how _you_ think things should be. MiniGo was doing fine on CGOS and going side by side with LeelaZero for a very long time. They then switched to 9x9 to iron some things out and if i've understood correctly are just now coming back to 19x19. To insult them, or to question their ability to create a strong bot is laughable. They have more than shown what they are capable of already, and anyone questioning that is not very knowledgable.

**Comment by hydrogenpi** [comment]
@MaxMaki I merely asked MiniGo how/where they were going to get the computing power from if they stated they aren't going the crowd distributed route, and they stated they want to start out the gate with a 20 block.... and their intent is to get to be a top bot in strength... Is it not allowed to ask the obvious question here?

**Comment by MaxMaki** [comment]
@hydrogenpi They've stated many many times already that they've had access to TPUs through Google and probobly will in the future as well. That's how they went side by side with LeelaZero for a long time (and actually infront of LeelaZero in strength) and also produced their 9x9 version... This discussion has already been had here on Github, search it.

**Comment by hydrogenpi** [comment]
So this is a Google employee, hosting his entire project on Google (even the self playing games), and getting their entire compute power from Google (via Cloud TPU), recreating Google's own work? Is this really consider "independent" in the academic sense or I can't help but feel for PR purposes its almost like tripping dipping, instead of letting someone else or independent recreate from their papers (like how LZ is doing) they seem to want to get their first (again) themselves in the form of MiniGo.

**Comment by MaxMaki** [comment]
@hydrogenpi  Maybe instead of speculating without any knowledge you'll go to their github and read and see how many bugs and problems they've had (commits, etc.), and how much info they've shared with this project as well about their different tests...

**Comment by MaxMaki** [comment]
And honestly who cares if they get there first? They're publishing all their games as well, as well as all their info!

**Comment by WesleyDeng** [comment]
BTW, the current best network 1ccb7342's final win% against the previous is 54.48%:
2018-03-19 17:20 | 1ccb7342 VS b7768081 | 237 : 198 (54.48%) | 435 / 400 | fail

**Comment by hydrogenpi** [comment]
@wesleyDeng, yes that is interesting that 1cc promoted at 54.48% and we are all looking at a 54.95% and haven't promoted it yet.

**Comment by Cabu** [comment]
@hydrogenpi then why 1ccb is 35MB and 089e is 76MB?

**Comment by nathanloop** [comment]
1cc got promoted because it went ahead early and got a PASS before falling behind later, I swear this comes up once a week. b09 did not do that. We're not at a full training window yet. A larger network is coming out soon. This doesn't really matter.

**Comment by hydrogenpi** [comment]
@cabu, they are both 12MB compressed. I loaded the 54.95% net into sabaki and gtp reports its cleary 10 block 128 filters.

**Comment by alreadydone** [comment]
@Cabu 089e [was](https://github.com/gcp/leela-zero/issues/1030#issuecomment-374947994) a 192x10 net trained with SWA (stochastic weight averaging) technique; GCP did such experiments on larger nets to not interfere with normal progress. We haven't switched to 192x10, so 128x10 nets continue to come out, and the subject of this discussion (b0963378) is one of them.

And this is a likely important change GCP decided to make: https://github.com/gcp/leela-zero/pull/1083

@hydrogenpi AlphaGo was DeepMind's work. DeepMind and Google are now [separate subsidiary companies](http://qr.ae/TU1yrW) of Alphabet Inc. (the Alphabet restructuring was probably not complete when AG played Lee Sedol since the matches were called Google DeepMind Challenge). DeepMind employees might have better chances to have access to the data (weights and code), but a group of Google employees has little other than the AGZ paper as their references.

**Comment by hydrogenpi** [comment]
@alreadydone thanks for the clarification. I had thought the restructuring was mainly for tax benefits like how Apple had restructured as well. In the past companies like Dropcam which were bought out by Nest went under the Google umbrella just to get seperated out into AlphaBet just to then get back into the Google nest (no pun intended) again. So its hard to say if one day Deepmind becomes Google again how that will change everything.

**Comment by gcp** [comment]
> but if MiniGo gets super strong and Google shares the weights with them, or in other ways helps them out under the table, I could see that not ending too well for LZ.

Sounds like a great ending to me. Just patch it up to work with the AGZ weights. Would save us a lot of work.

**Comment by gcp** [comment]
> I fear that if only a week's stall is already testing the patience of our users, we will be in for a lot of trouble in the near future. Because the next network may take a couple of weeks, and game generation will only be slower with larger networks.

I've said a few times that doing a full 256x20 or 256x40 run to the same level as AlphaGo is somewhat unfeasible in our current setup because there will be too much attrition when the progress inevitably starts stalling. To some extent this thread is demonstrating that.

**Comment by gcp** [comment]
I'm going to close this WONTFIX because (as you can probably guess) I have no plans of faking a promotion to keep impatient people happy.

**Comment by gcp** [comment]
>FIrst of all, as far as I currently know, we haven't heard definitively from gcp if he ultimately has decided whether or not this run will be the first of many, or if the intent is this run will be the last. I think knowing that changes a lot of the calculus of things.

The goal was to run to 128x6, demonstrate the the setup works and the AGZ results are reproducible and can be replicated by a distributed project, and then go away. As I said above, I think a full 256x20 or x40 run require some changes that I'm both not interested and not very well qualified to handle, especially on the community side. I can't speak for @roy7 on the server side of things but for sure there's going to be a moment where this turns from a fun hobby to unfun work.

What led me to reconsider was that doing tests on 128x10 gave a big jump, and in the mean time LZ surpassed a lot of existing programs. Pulling the plug when you're pushing the limit of what is freely available would've been...well, maybe "cruel" is a good word!

There's enough obvious things to try that I doubt we will end up being "stalled" for long at this level, but for sure we will eventually stall somewhere.

**Comment by hydrogenpi** [comment]
@gcp Okay thanks for the explaination it makes sense. I think it is inevitable that everything will come to a stall somewhere, even AGZ made little to no progress after its first 20 days or so. But like as you mentioned, its good that you did not stop the experienment just as it was getting to pro level and became one of the strongest out there freely available, and certainly the first to get to pro level with open source and using non-supervised learning (vs AQ is supervised for example)  I think you are probably right that it still has room to grow, I just hope that it can approach superhuman levels before the final "stall"... so that there can be freely available a superhuman bot for the entire world to use. 

My two cents on the Google situation, I'm grateful for their publish of the two papers, without which it almost certaintly will be the case that today there wouldn't have been free open source zero bot (LZ) of pro levels. And I'm realistic and cognizant of the fact that as a company they have the perojative to do things in their own best interest and do not "owe" the Go community anything at all. Having said that, I recall in the past the developers stated that the reason they choose not to go the open source route with the code was because it would be difficult to seperate out the AG code from their own internal codebase and they didn't wish to leak any intellectual property etc... But now that officially AG/AGZ has finally retired, I don't see the reason as to why Google would not release the weights themselves. Not talking about training code even, just the final raw weights, or if they didn't want to give their crown jewel away, even intermediate weights or just the quality self playing games etc... This I find puzzling.

I know this is the LZ github channel but I just wanted to make one final remark with regards to the MiniGo. So my current understanding is that MiniGo is lead by Andrew Jackson, an employee of Google, and while he has taken care to disclaimer the seperation of the MiniGo project from the Deepmind team, it seems apparent to me that everything related to MiniGo project is being done on Google arch, including all the self playing games hosted on Google storage (which requires a Google account to access) and I was also told that since they have refused the crowd distributed computing route of asking the public for help, that MiniGo will rely exclusively on Google's TPU cloud, aka Google will also provide the means of compute power for the project. Andrew Jackson stated on many occasions that his intent is for MiniGo to become a "top bot" and I can also surmise that to mean a superhuman level bot stronger than any pro. 

So to me, the 800 pound elephant in the room is why are we duplicating so much duplicated effort? We know Deepmind/Google already achieved great success with their zero method. I'm sure no one thinks Google didn't achieve their results fair and square, unlike things like cold fusion in a jar that was never able to be duplicated, I believe and I think we all believe Deepmind paper to be legit, I understand the scientific satisfication of being able to nonetheless independently duplicate and verify the results and to produce the data etc etc... We have on one hand Google /Deepmind who already first did the work, computed the weights, made a superhuman world class Go AI, but now we have multiple different Go AI projects (LZ, MiniGo, just to name a few) who wishes to each independently reproduce Google's work, to me it just seems like dual effort, since 1) we already know it can be done 2) what is the point of duplicating it twice and moreover, triply ironic is that MiniGo is basically Google trying to prove their work to themselves, I'm thinking is the right hand really not talking to the left hand?!
